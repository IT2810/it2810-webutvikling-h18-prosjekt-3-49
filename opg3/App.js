import React from 'react';
import {StyleSheet, View, KeyboardAvoidingView} from 'react-native';
import Calendar from './components/Calendar.js';
import ContactManager from './components/Contacts.js';
import Goals from './components/Goals.js';
import Accelerometer from './components/Accelerometer';
import { createDrawerNavigator } from 'react-navigation';

class CalendarScreen extends React.Component {
    render() {
        return (
            <KeyboardAvoidingView behavior='padding' enabled>
                <Calendar />
            </KeyboardAvoidingView>
        );
    }
}

class ContactsScreen extends React.Component {
    render() {
        return (
            <KeyboardAvoidingView style={styles.container} behavior='padding' enabled>
                <ContactManager />
            </KeyboardAvoidingView>
        )
    }
}

class GoalsScreen extends React.Component {
    render() {
        return(
            <KeyboardAvoidingView style={styles.container} behavior='padding' enabled>
                <Goals />
            </KeyboardAvoidingView>
        )
    }
}

class AccelerometerScreen extends React.Component {
    render() {
        return(
            <View style={styles.container, styles.centered}>
                <Accelerometer/>
            </View>
        )
    }
}

export default createDrawerNavigator({
    Calendar: {
        screen: CalendarScreen,
    },
    Contacts: {
        screen: ContactsScreen,
    },
    Goals: {
        screen: GoalsScreen,
    },
    Accelerometer: {
        screen: AccelerometerScreen,
    },
});

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
        width: '70%',
        flex: 1,
        justifyContent: 'center',
        alignSelf: 'center',
    }
});
