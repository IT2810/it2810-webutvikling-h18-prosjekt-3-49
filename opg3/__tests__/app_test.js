import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../App';

const renderer = new ShallowRenderer();
renderer.render(<App/>);
const result = renderer.getRenderOutput();

it('renders correctly', () => {
    expect(result.type).toBe('div');
});
