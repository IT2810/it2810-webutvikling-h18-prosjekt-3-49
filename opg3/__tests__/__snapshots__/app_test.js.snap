// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shallow correctly 1`] = `
<Navigator
  navigation={
    Object {
      "actions": Object {
        "closeDrawer": [Function],
        "openDrawer": [Function],
        "toggleDrawer": [Function],
      },
      "addListener": [Function],
      "closeDrawer": [Function],
      "dangerouslyGetParent": [Function],
      "dispatch": [Function],
      "getChildNavigation": [Function],
      "getScreenProps": [Function],
      "goBack": [Function],
      "isFocused": [Function],
      "navigate": [Function],
      "openDrawer": [Function],
      "router": Object {
        "childRouters": Object {
          "Accelerometer": null,
          "Calendar": null,
          "Contacts": null,
          "Goals": null,
        },
        "getActionCreators": [Function],
        "getActionForPathAndParams": [Function],
        "getComponentForRouteName": [Function],
        "getComponentForState": [Function],
        "getPathAndParamsForState": [Function],
        "getScreenOptions": [Function],
        "getStateForAction": [Function],
      },
      "setParams": [Function],
      "state": Object {
        "closeId": 0,
        "index": 0,
        "isDrawerOpen": false,
        "isTransitioning": false,
        "openId": 0,
        "routes": Array [
          Object {
            "key": "Calendar",
            "params": undefined,
            "routeName": "Calendar",
          },
          Object {
            "key": "Contacts",
            "params": undefined,
            "routeName": "Contacts",
          },
          Object {
            "key": "Goals",
            "params": undefined,
            "routeName": "Goals",
          },
          Object {
            "key": "Accelerometer",
            "params": undefined,
            "routeName": "Accelerometer",
          },
        ],
        "toggleId": 0,
      },
      "toggleDrawer": [Function],
    }
  }
/>
`;
